src/test/kotlin/org/rust/lang/core/type/RsGenericExpressionTypeInferenceTest.kt

```diff
@@ -241,6 +241,96 @@ class RsGenericExpressionTypeInferenceTest : RsTypificationTestBase() {
         }
     """)
 
+    fun `test self 1`() = testExpr("""
```

Comment:
We still did not have plain `self` param tests? I'm failed to find them, but it looks really weird...